collector/clickhouse.go

```diff
@@ -71,18 +68,10 @@ func (ch *Clickhouse) connect() error {
 		return fmt.Errorf("failed to parse Clickhouse DSN: %w", err)
 	}
 
-	var chTLS *tls.Config
-	enableTLS := !ch.opts.DisableTLS
-	if enableTLS {
-		chTLS = &tls.Config{
-			InsecureSkipVerify: true, //nolint:gosec
-		}
-	}
-
 	ch.conn, err = clickhouse.Open(&clickhouse.Options{
 		Addr: options.Addr,
 		Auth: options.Auth,
-		TLS:  chTLS,
+		TLS:  options.TLS,
 		Debugf: func(format string, v ...interface{}) {
 			ch.log.Infof("Clickhouse debug: "+format, v...)
 		},
```

Comment:
Would be simpler (and will preserve all other options supported in DSN, including `skip_verify`):

```go
	options.Debugf = func(format string, v ...interface{}) {
		ch.log.Infof("Clickhouse debug: "+format, v...)
	}
	ch.conn, err = clickhouse.Open(options)